src/lib/builders/slider/create.ts

```diff
@@ -294,31 +320,38 @@ export const createSlider = (props?: CreateSliderProps) => {
 	});
 
 	const tick = builder(name('tick'), {
-		stores: [ticks, value, min, max, step, orientation],
-		returned: ([$ticks, $value, $min, $max, $step, $orientation]) => {
+		stores: [ticks, value, min, max, step, orientation, dir],
+		returned: ([$ticks, $value, $min, $max, $step, $orientation, $dir]) => {
 			let index = -1;
 			return () => {
 				index++;
 
-				const horizontal = $orientation === 'horizontal';
+				// The track is divided into sections of ratio `step / (max - min)`
+				const tickPosition = `${index * ($step / ($max - $min)) * 100}%`;
+
+				// Offset each tick by -50% to center it, except the first and last ticks.
+				// The first tick is already positioned at the start of the slider.
+				// The last tick is offset by -100% to prevent it from being rendered outside.
+				const isFirst = index === 0;
+				const isLast = index === $ticks - 1;
+				const offsetPercentage = isFirst ? 0 : isLast ? -100 : -50;
+
 				const style: Record<string, string | number | undefined> = {
 					position: 'absolute',
 				};
 
-				// The track is divided into sections of ratio `step / (max - min)`
-				const positionPercentage = index * ($step / ($max - $min)) * 100;
-				style[horizontal ? 'left' : 'bottom'] = `${positionPercentage}%`;
-
-				// Offset each tick by half its size to center it, except for
-				// the first tick as it would be rendered outside the slider.
-				//
-				// As for the last tick, offset it by its full size rather than
-				// half also to prevent it from being rendered outside.
-				if (index === $ticks - 1) {
-					// Left is negative, down is positive.
-					style.translate = horizontal ? '-100% 0' : '0 100%';
-				} else if (index !== 0) {
-					style.translate = horizontal ? '-50% 0' : '0 50%';
+				if ($orientation === 'horizontal' && $dir === 'rtl') {
```

Comment:
These ifs can be a bit complex too, something similar to what i've shown before would be useful 😄

Reply:
> These ifs can be a bit complex too, something similar to what i've shown before would be useful 😄

I'm not the biggest fan of ternaries, personally, but I'll fix the ifs. I wish JS had native pattern matching similar or switch statements with multiple values.

Reply:
Hey! So I just pushed the changes. I chose a different approach from ternary statements. If you still prefer ternaries, I have no problem refactoring the code, but I think the new changes are much more readable than both the ternaries and the if statements.

Reply:
I do prefer the ternaries, but the direction derived with the switch case is pretty nice too 😄 You can keep it!